WordPress/src/main/java/org/wordpress/android/ui/posts/InsertMediaDialog.java

```diff
@@ -0,0 +1,305 @@
+        return mGalleryType;
+    }
+
+    private void setGalleryType(@NonNull GalleryType galleryType) {
+        if (galleryType == mGalleryType) {
+            return;
+        }
+
+        mGalleryType = galleryType;
+
+        // column count applies only to thumbnail grid
+        boolean showNumColumns = (galleryType == GalleryType.DEFAULT);
+        if (showNumColumns && mNumColumnsContainer.getVisibility() != View.VISIBLE) {
+            AniUtils.fadeIn(mNumColumnsContainer, AniUtils.Duration.SHORT);
+        } else if (!showNumColumns && mNumColumnsContainer.getVisibility() == View.VISIBLE) {
+            AniUtils.fadeOut(mNumColumnsContainer, AniUtils.Duration.SHORT, View.INVISIBLE);
+        }
+
+        @IdRes final int resId;
+        @DrawableRes final int drawableId;
+        switch (galleryType) {
+            case CIRCLES:
+                resId = R.id.radio_circles;
+                drawableId = R.drawable.gallery_icon_circles;
+                break;
+            case SLIDESHOW:
+                resId = R.id.radio_slideshow;
+                drawableId = R.drawable.gallery_icon_slideshow;
+                break;
+            case SQUARES:
+                resId = R.id.radio_squares;
+                drawableId = R.drawable.gallery_icon_squares;
+                break;
+            case TILED:
+                resId = R.id.radio_tiled;
+                drawableId = R.drawable.gallery_icon_tiled;
+                break;
+            default:
+                resId = R.id.radio_thumbnail_grid;
+                drawableId = R.drawable.gallery_icon_thumbnailgrid;
+                break;
+        }
+
+        RadioButton radio = (RadioButton) mGalleryRadioGroup.findViewById(resId);
+        if (!radio.isChecked()) {
+            radio.setChecked(true);
+        }
+
+        // scale out the gallery type image, then set the new image and scale it back in
+        final ImageView imageView = (ImageView) getView().findViewById(R.id.image_gallery_type);
+        AniUtils.scaleOut(imageView, View.VISIBLE, AniUtils.Duration.SHORT, new AniUtils.AnimationEndListener() {
+            @Override
+            public void onAnimationEnd() {
+                imageView.setImageResource(drawableId);
+                AniUtils.scaleIn(imageView, AniUtils.Duration.SHORT);
+            }
+        });
+    }
+
+    public int getNumColumns() {
+        return mNumColumns;
+    }
+
+    private void setNumColumns(int numColumns, boolean fromSeekBar) {
+        // seekbar is zero-based, so increment the column count if this was called from it
+        if (fromSeekBar) {
+            mNumColumns = numColumns + 1;
+        } else {
+            mNumColumns = numColumns;
+            mNumColumnsSeekBar.setProgress(numColumns);
+        }
+
+        TextView textValue = (TextView) getView().findViewById(R.id.text_num_columns_label);
+        textValue.setText(String.format(getString(R.string.media_gallery_column_count), mNumColumns));
```

Comment:
We should probably add a switch and another string resource for pluralization here - this currently will say `1 columns`.